test/client_test.dart

```diff
@@ -239,6 +239,7 @@ void main() {
           serverWriter.writeDataFrame(invalidStreamId, [42]);
 
           // Make sure the client sends a [RstStreamFrame] frame.
+          expect(await nextFrame() is WindowUpdateFrame, true);
```

Comment:
Consider also checking that the `streamId == 0`, i.e. this is a window update frame for the connection and not for the stream.

nit: Consider consistently using `isA<XXX>()` or `is XXX` - see line below. (also other places)